PurchasesCoreSwift/Networking/HTTPClient.swift

```diff
@@ -0,0 +1,306 @@
+                } else {
+                    Logger.error("Could not create request to \(path) without body")
+                }
+                if let completionHandler = maybeCompletionHandler {
+                    self.operationDispatcher.dispatchOnMainThread {
+                        completionHandler(-1, nil, ErrorUtils.networkError(withUnderlyingError: ErrorUtils.unknownError()))
+                    }
+                }
+                return
+            }
+
+            let request = HTTPRequest(httpMethod: httpMethod,
+                                      path: path,
+                                      requestBody: maybeRequestBody,
+                                      authHeaders: authHeaders,
+                                      retried: retried,
+                                      urlRequest: urlRequest,
+                                      completionHandler: maybeCompletionHandler)
+
+            if performSerially && !retried {
+                if self.currentSerialRequest != nil {
+                    let logMessage =
+                        String(format: Strings.network.serial_request_queued, self.queuedRequests.count, httpMethod, path)
+                    Logger.debug(logMessage)
+                    self.queuedRequests.append(request)
+                    return
+                } else {
+                    Logger.debug(String(format: Strings.network.starting_request, httpMethod, path))
+                    self.currentSerialRequest = request
+                }
+            }
+
+            let logMessage = String(format: Strings.network.api_request_started,
+                                    urlRequest.httpMethod ?? "",
+                                    urlRequest.url?.path ?? "")
+            Logger.debug(logMessage)
+
+            let task = self.session.dataTask(with: urlRequest) { (data, urlResponse, error) -> Void in
+                self.handleResponse(urlResponse: urlResponse,
+                                    request: request,
+                                    data: data,
+                                    error: error,
+                                    completionHandler: maybeCompletionHandler,
+                                    beginNextRequestWhenFinished: performSerially,
+                                    retried: retried)
+            }
+            task.resume()
+        }
+    }
+
+    // swiftlint:disable function_parameter_count
+    func handleResponse(urlResponse maybeURLResponse: URLResponse?,
+                        request: HTTPRequest,
+                        data maybeData: Data?,
+                        error maybeNetworkError: Error?,
+                        completionHandler maybeCompletionHandler: ((Int, [AnyHashable: Any]?, Error?) -> Void)?,
+                        beginNextRequestWhenFinished: Bool,
+                        retried: Bool) {
+        operationDispatcher.dispatchOnHTTPSerialQueue { [self] in
```

Comment:
This could be extracted into a new function so you could do this:
`operationDispatcher.dispatchOnHTTPSerialQueue { self.internalHandleResponse(...) }`
That way, the business logic is at the root of the scope instead of indented. Having a function that is a huge body with indentation makes reasoning about the logic more difficult, even if it's just 1 extra scope.

Reply:
I actually really liked the `threadUnsafeFoo...` naming from device cache, I'm gonna emulate that